framework/src/base/Attributes.C

```diff
@@ -309,21 +309,38 @@ AttribPreIC::isEqual(const Attribute & other) const
   return isMatch(other);
 }
 
-void
-AttribPreAux::initFrom(const MooseObject * /*obj*/)
-{
-}
 bool
 AttribPreAux::isMatch(const Attribute & other) const
 {
   auto a = dynamic_cast<const AttribPreAux *>(&other);
-  return a && (a->_val == _val);
+  if (!a || a->_vals.size() < 1)
+    return false;
+  auto cond = a->_vals[0];
+  if (cond == Moose::ALL)
+    return true;
+
+  for (auto val : _vals)
+    if (val == Moose::ALL || val == cond)
+      return true;
+  return false;
 }
 
 bool
 AttribPreAux::isEqual(const Attribute & other) const
 {
-  return isMatch(other);
+  auto a = dynamic_cast<const AttribPreAux *>(&other);
+  if (!a || a->_vals.size() != _vals.size())
+    return false;
+
+  for (size_t i = 0; i < a->_vals.size(); i++)
+    if (a->_vals[i] != _vals[i])
+      return false;
```

Comment:
What if they have the same entries, but are in a different order?

Reply:
> But this code promotes a UO to pre aux for _all_ execute on flags if an aux kernel running on only _one_ execute on flag depends on that uo. Is this what you wanted? Or do we need to revisit the implementation to try to achieve the full handling I just described?

I don't think this is the case, the test I added I think shows this does not occur. In the test I added, the section of input shown below is the important part. Before the code changes proposed in this MR, the `scale1` UO is forced into the `PRE_AUX` group because `NormalizationAuxTest1` depends on `scale1`. But `NormalizationAuxTest1` only execs on `FINAL` so there is not a dependency for any other exec flag. When I run this test with the code in this MR, I see that `scale1` is not run in `PRE_AUX` except for `FINAL`

```
[AuxKernels]
  active = 'NormalizationAuxTest1'
  [NormalizationAuxTest1]
    type = NormalizationAux
    variable = w
    source_variable = u
    normalization = scale1
    execute_on = 'FINAL'
  []
  ...
  [Postprocessors]
  ...
  [./scale1]
    type = ScalePostprocessor
    value = total_u
    scaling_factor = 1
  [../]
```

Reply:
I copied this logic from `AttribExecOns::isEqual(const Attribute & other) const` but I'm not sure why `isEqual` needs to check for identical ordering. Do you now why `AttribExecOns` checks for ordering? If there is no reason, then I think changing this to not check the ordering makes sense.

Reply:
I haven't changed this in my most recent code updates. I still want `AttribPreAux` and `AttribPostAux` to be similar to `AttribExecOns` which uses a `_vals` a vector and includes this logic to check `isEqual`. However, if you think I should change `_vals` to a std::set for both `AttribPreAux` and `AttribPostAux`, I can change the logic of `AttribPreAux::isEqual` and  `AttribPostAux::isEqual` to check if the sets are the same

Reply:
Yes - this needs to be order independent.  AttribExecOns is a bit different in that its vector is only ever populated internally inside its initFrom function - where it retrieves the exec on flags from an ordered source.  In the code here, however, the pre/post aux flags are inserted externally - i.e. passed in via a constructor arg - so the only way they are ordered is if the calling code knows to only pass in an ordered vector - which is much less safe.  So this needs to be sorted.